internal/provider/newrelic/provider_integration_test.go

```diff
@@ -162,3 +216,21 @@ func newrelicProviderWithMetric(t *testing.T, metric newrelic.Metric) provider.E
 
 	return p
 }
+
+func runProxy(t *testing.T, port int, f func(string, *goproxy.ProxyCtx) (*goproxy.ConnectAction, string)) *http.Server {
+	t.Helper()
+
+	proxy := goproxy.NewProxyHttpServer()
+	proxy.OnRequest(goproxy.ReqHostMatches(regexp.MustCompile(".*"))).HandleConnectFunc(f)
```

Comment:
just out of curiosity, what means the  `string` parameter and return value in the `f` function?

Reply:
Updated in 34b65a2a2eaaa01c9382813cd910343c97f7c1dc